turbomoleio/core/molecule.py

```diff
@@ -511,6 +604,419 @@ def from_dict(cls, d):
+                coord file or any format supported in pymatgen Molecule.
+
+        Returns:
+            An instance of MoleculeSystem.
+        """
+
+        if fmt == "coord":
+            dg = DataGroups(string=string)
+            coordinates_str = dg.sdg("$coord", strict=True)
+            if not coordinates_str:
+                raise ValueError("The string does not contain $coord!")
+            cell_str = dg.sdg("$cell", strict=True)
+            lattice_str = dg.sdg("$lattice", strict=True)
+            periodic_str = dg.sdg("$periodic", strict=True)
+            if periodic_str is None:
+                raise ValueError('The $periodic data group should be set for periodic systems.')
+            if cell_str is None and lattice_str is None:
+                raise ValueError('The $cell or $lattice data group should be set for periodic systems.')
+            if cell_str is not None and lattice_str is not None:
+                raise ValueError('Only one of $cell and $lattice data group should be set for periodic systems.')
+            mol, fi = get_mol_and_indices_frozen(coordinates_str, cell_string=cell_str,
+                                                 lattice_string=lattice_str,
+                                                 periodic_string=periodic_str,
+                                                 periodic_extension=periodic_extension)
+            periodicity = int(periodic_str.strip())
+
+            int_def_str = dg.sdg("$intdef", strict=True)
+            if int_def_str is not None:
+                raise ValueError('Internal definitions for periodic systems is not supported.')
+
+            user_def_bonds_str = dg.sdg("$user-defined bonds", strict=True)
+            if user_def_bonds_str is not None:
+                raise ValueError('User-defined bonds for periodic systems is not supported.')
+
+            return cls(mol, frozen_indices=fi, periodicity=periodicity)
+
+        else:
+            return cls(Structure.from_str(string, fmt))
+
+    @classmethod
+    def from_file(cls, filepath, fmt=None, periodic_extension=5.0):
+        """
+        Creates an instance from a file. Could be a coord file or any other
+        format supported by pymatgen Molecule.
+
+        Args:
+            filepath (str): path to the file.
+            fmt (str): the format of the data. could be "coord" for Turbomole
+                coord file or any format supported in pymatgen Molecule.
+                If None the code would try to infer it from the file name.
+
+        Returns:
+            An instance of MoleculeSystem.
+        """
+        fname = os.path.basename(filepath)
+        if fmt == "coord" or (fmt is None and fnmatch(fname, "coord")):
+            with open(filepath) as f:
+                return cls.from_string(f.read(), fmt="coord", periodic_extension=periodic_extension)
+        else:
+            return cls(molecule=Structure.from_file(filepath))
+
+    def to_coord_file(self, filepath="coord"):
```

Comment:
maybe some of these methods could go in the base class?

Reply:
Done, I've moved some of these in the BaseSystem.